### PR TITLE
Fix Windows example

### PR DESCRIPTION
**Pull request by rcurtin**
I fired up a Windows 10 VM (wow that install process is naggy and full of advertising), and installed Visual Studio, and made some changes so that the example in `doc/examples/sample-ml-app` works with the latest version of Visual Studio.  These changes come from [a comment on the 4.0.0 release PR](https://github.com/mlpack/mlpack/pull/3286#discussion_r999877355).

I also added a README detailing the system setup that you have to do first to make that example work (it's not too hard, just download some libraries and put them in place...).

There were also some very basic code changes to adapt to the flattened `mlpack::` namespace.

Because it is Windows, the files now have Windows newlines, which makes the diff a little hard to read, unfortunately.  Here are the changes:

 * We now require include directories for cereal, ensmallen, Armadillo, and mlpack.
 * We link against OpenBLAS and copy it into place for install.
 * The C++ standard is changed to C++17 (which is required for it to build on Windows due to poor C++14 support).
 * The `/Zc:__cplusplus` option needs to be added so that MSVC sets `__cplusplus` [correctly](https://learn.microsoft.com/en-us/cpp/build/reference/zc-cplusplus?view=msvc-170).
 * Superfluous `using namespace` directives are removed from the source file.

### CODE REVIEW

File: doc/examples/sample-ml-app/README.txt [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+
+In short, you must download mlpack and four libraries, and install the sources
+and library files into C:\mlpack\.
+
+ * OpenBLAS: https://github.com/xianyi/OpenBLAS/releases/download/v0.3.21/OpenBLAS-0.3.21-x64.zip
+   Download the .zip, and extract it into C:\mlpack\openblas-0.3.21\
+
+ * Armadillo: https://mlpack.org/files/armadillo-11.4.1.tar.gz
+   Download the .tar.gz, and extract it into C:\mlpack\armadillo-11.4.1\; note
+   that you may need to use a program such as 7Zip ( ) to unpack this archive.
```
Comment (zoq) [inline doc/examples/sample-ml-app/README.txt:15]: `( )` is that a link you wanted to add?

Comment (rcurtin) [reply doc/examples/sample-ml-app/README.txt]: Thank you!  This is what code review is for I guess. :)  Link fixed in 5d1190a.

**Review by shrit — APPROVED** [review]
LGTM

**Review by zoq — APPROVED** [review]
No further comments from my side, thanks.

**Pull request merged**